chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsStateDifferTest.kt

```diff
@@ -0,0 +1,65 @@
+package com.pusher.chatkit.rooms.state
+
+import assertk.assertThat
+import assertk.assertions.containsExactly
+import com.pusher.chatkit.state.LeftRoom
+import com.pusher.chatkit.state.ReconnectJoinedRoom
+import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.state.State
+import io.mockk.every
+import io.mockk.mockk
+import org.reduxkotlin.GetState
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class JoinedRoomsStateDifferTest : Spek({
+
+    describe("given one room") {
+        val state = mockk<GetState<State>>(relaxed = true)
+        val differ = JoinedRoomsStateDiffer(state)
+
+        every { state().joinedRoomsState } returns JoinedRoomsState(
+            rooms = mapOf(roomOneId to roomOne),
+            unreadCounts = mapOf(roomOneId to 1))
+
+        describe("when room is joined") {
+            val actions = differ.toActions(
+                newRooms = listOf(roomOne, roomTwo),
+                newUnreadCounts = mapOf(roomOneId to 1, roomTwoId to 2))
+
+            it("then the result contains JoinedRoom action") {
+                assertThat(actions).containsExactly(ReconnectJoinedRoom(room = roomTwo, unreadCount = 2))
+            }
+        }
+    }
+
+    describe("given two rooms") {
+        val state = mockk<GetState<State>>(relaxed = true)
+        val differ = JoinedRoomsStateDiffer(state)
+
+        every { state().joinedRoomsState } returns JoinedRoomsState(
+            rooms = mapOf(roomOneId to roomOne, roomTwoId to roomTwo),
+            unreadCounts = mapOf(roomOneId to 1, roomTwoId to 2))
```

Comment:
there's still partial mocking here and formatting issue
```suggestion
        val state = mockk<GetState<State>>()
        every { state().joinedRoomsState } returns JoinedRoomsState(
            rooms = mapOf(roomOneId to roomOne, roomTwoId to roomTwo),
            unreadCounts = mapOf(roomOneId to 1, roomTwoId to 2)
        )
        val differ = JoinedRoomsStateDiffer(state)
```
also for better readability it's best to keep mock setup in one place

Reply:
I don't think you intended to remove the JoinedRoomsStateDiffer in this commit - instead I have moved the mocks closer to the object for readability in both given sections.

Reply:
oh yes, sorry my original version had that and then I removed the differ in mistake thinking I pasted one line to much from my IDE ;)